core/src/main/java/pl/agh/edu/utils/CustomBigDecimal.java

```diff
@@ -0,0 +1,92 @@
+package pl.agh.edu.utils;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.util.Arrays;
+import java.util.Comparator;
+
+import ch.obermuhlner.math.big.BigDecimalMath;
+
+public class CustomBigDecimal {
```

Comment:
jeszcze jak teraz robiłem testy e2e to zauważyłem że może nam rozwalić sporo jak wyjdziemy powyżej 999T może warto wprowadzić ograniczenie w metodach toString() i roundToStringValue()